### PR TITLE
On login we create a default calendar and a default addressbook for t…

### LINKED ISSUE #22360: Create default calendar and address book via login-hook
The login hook should trigger a mechanism to check whether the user has a calendar or an address book. If there is none present, it should create one.

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…he user - fixes #22360 

@georgehrke @nickvergessen @PVince81

### CODE REVIEW

**Comment by nickvergessen** [comment]
is this really required? Or should we do this on a request to the dav endpoints only?

File: apps/dav/lib/hookmanager.php [hunk 1]
```diff
@@ -80,4 +97,19 @@ public function changeUser($params) {
...
 		$this->syncService->updateUser($user);
 	}
+
+	public function postLogin($params) {
+		$user = $this->userManager->get($params['uid']);
+
+		$principal = 'principals/users/' . $user->getUID();
+		$calendars = $this->calDav->getCalendarsForUser($principal);
+		if (empty($calendars)) {
+			$this->calDav->createCalendar($principal, 'default', []);
```
Comment (PVince81) [inline apps/dav/lib/hookmanager.php]: does this throw an exception if the calendar already exists ? You know... race conditions, two concurrent logins.

Comment (DeepDiver1975) [inline apps/dav/lib/hookmanager.php]: > does this throw an exception if the calendar already exists ? You know... race conditions, two concurrent logins.

good point .... will adopt

**Comment by DeepDiver1975** [comment]
> Or should we do this on a request to the dav endpoints only?

using a webdav login hook? we would need to introduce one

**Comment by nickvergessen** [comment]
More like "the first webdav request to such a resource", but I see this escalating...
So should be okay, once fixed

**Comment by nickvergessen** [comment]
Tested and works :+1:

**Comment by rullzer** [comment]
Works as described :) :+1:

**Pull request merged**